### PR TITLE
[NS-956] - updating vcluster, configured values for sqlite and adjust…

### PR DESCRIPTION
**Pull request by drew-viles**
Upgrading cluster and enforcing the embedded sqlite backend.
The previous version used etcd (even though it [defaulted to sqllite](https://www.vcluster.com/docs/v0.19/architecture/control_plane/) ¯\_(ツ)_/¯) whereas now we explictly define it. It does [default to it now](https://www.vcluster.com/docs/vcluster/configure/vcluster-yaml/control-plane/components/backing-store/), but with there being other options available, then should something change (as you can see something has already below with the clusterConfig name) this will just ensure it sticks to it. **BREAKING CHANGE - No migration path supported**

[Replaced](https://github.com/loft-sh/vcluster/blob/v0.20/config/legacyconfig/migrate.go#L119-L121) `autoDeletePersistentVolumeClaims` with `controlPlane.statefulSet.persistence.volumeClaim.retentionPolicy`.

`hostpathMapper` value has been relocated due to some restructuring around the charts - also some of the functionality of it (a centralised option) has been put behind the paywall - but that shouldn't affect us.

Updated the slice string reference within the `config.Cluster` to `kubernetes` as the name in the config has (helpfully) changed. **This would likely be a breaking change but since we have two already and have discussed blowing away vcluster altogether to roll out fresh, let's ignore it and move on with our lives.**

This will switch vcluster from k3s to k8s which is the new recommended default for vcluster. We can stick with k3s if we'd prefer though. However, if we're going to blow everything away and make the switch now, then let's do it in this release. **BEAKING CHANGE - Switching distro not supported**

### CODE REVIEW

File: pkg/provisioners/helmapplications/vcluster/provisioner.go [hunk 1]
```diff
@@ -19,7 +19,6 @@ package vcluster
 
 import (
 	"context"
```
Comment (spjmurray) [inline pkg/provisioners/helmapplications/vcluster/provisioner.go:21]: This is a legitimate error because you've deleted something you shouldn't have...

**Review by spjmurray — CHANGES_REQUESTED** [review]

File: pkg/provisioners/helmapplications/vcluster/provisioner.go (hunk 1 above)
Comment (drew-viles) [reply pkg/provisioners/helmapplications/vcluster/provisioner.go:21]: friggin IDE!

Comment (drew-viles) [reply pkg/provisioners/helmapplications/vcluster/provisioner.go:21]: Wait, am I missing something? Itt's only a new line been removed? Is there a formatting requirement here?

**Comment by spjmurray** [comment]
Okay, so that hostPathMapper is due to....

```
[Thu 17 Oct 10:05:05 BST 2024] simon@symphony ~/src/github.com/unikorn-cloud/kubernetes git show b3f5740e
commit b3f5740e964b95fe818076175eca5ab0ce0ec935
Author: Simon Murray <spjmurray@yahoo.co.uk>
Date:   Mon Jun 5 14:57:10 2023 +0100

    Enable Virtual Cluster Path Mapper (#233)
    
    Apparently some eBPF trickery is going on to intercept access/stat/open
    calls and route them to the right places, so you can now run promtail
    inside a virtual cluster.
```

So by removing it, or making it a paid for feature, we'll lose logging ability...

**Comment by drew-viles** [comment]
Bugger.

Yeah This is on their docs:

> This feature is an extension to the existing [hostpath mapper](https://www.vcluster.com/docs/v0.19/o11y/logging/hpm) component of vCluster and requires an active vCluster.Pro license.
I'll tak ea look into [this](https://github.com/loft-sh/vcluster-hostpath-mapper?_gl=1*1v4p3bo*_gcl_au*MTQyOTMxMTA3Ni4xNzI5MDg0OTM3*_ga*MTAxNjEwNjA1OC4xNzI5MDg0OTM3*_ga_4RQQZ3WGE9*MTcyOTE1NjEwMy4zLjEuMTcyOTE1NjExNy40Ni4wLjE4OTg1NzE4NjA) then as it may be a case they don't supply it in the main vcluster deployment anymore but they do let you manage it yourself still.

I'm not sure how it works yet to be honest so research required.

File: pkg/provisioners/helmapplications/vcluster/provisioner.go (hunk 1 above)
Comment (drew-viles) [reply pkg/provisioners/helmapplications/vcluster/provisioner.go]: ok, evidently it is!! 

Christ, stupid IDE formatting - ok resolving this. I'll still get a linter bump pushed in across all repos as it doesn't hurt either way

**Comment by drew-viles** [comment]
Hang ong Hang on - I think they've actually got two now which is a centralised one and the "open source" one. 

It looks like they've hidden it all away in the `[values.schema.json](https://github.com/loft-sh/vcluster/blob/v0.20.1/chart/values.schema.json#L1619-L1629)` file and make no reference to is anywhere in the main values, basically you have to go hunting for it to find the setting. Let me test with that setting.

**Comment by drew-viles** [comment]
Nope, that's just a [ref](https://github.com/loft-sh/vcluster/blob/v0.20.1/chart/values.schema.json#L88) 🤦  Bastardo.

**Comment by drew-viles** [comment]
It still seems like ["something" is supported](https://www.vcluster.com/docs/vcluster/configure/vcluster-yaml/control-plane/components/host-path-mapper#vcluster-open-source) though... I'm just gonna try it, sod it.

**Comment by drew-viles** [comment]
ok... it's yet another minor change they've done that breaks things. So it turns out it is still supported. I've just been through what we had before and what exists now and it's a single character capitalisation difference.... because why not? 

Anyway, looks like it is supported but there are "pro" features in the way it deploys the daemonset. Basically they OpenSource version gets one per vcluster where as in the pro one you can have a centralised one!

**Comment by spjmurray** [comment]
In this context "pro" meaning a single point of failure?

**Comment by drew-viles** [comment]
ok @spjmurray testing is done.
Everything falls apart no matter how many approaches I take to upgrading it - it's a case of blowing it away and rebuilding as we suspected it would be as the multiple changes that will happen just aren't supported as migration paths with vcluster.

Udpated readme above to reflect all of this.

**Review by spjmurray — APPROVED** [review]
Seems to do what it claims to!

**Pull request merged**